.github/workflows/cicd.yml

```diff
@@ -178,17 +178,27 @@ jobs:
 
     needs: [lint, mypy, docs, packaging]
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 30
+    continue-on-error: ${{ matrix.optional }}
```

Comment:
@adiroiban I found the `continue-on-error` [docs](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) have an example showing how to have a job not stop the workflow if it fails, so used that for `trunk`, which leaves `fail-fast` as something that we can turn on or off independently.